README.md
Join my discord https://discord.gg/DfW4y6tNVn for the Nebula Nuker discord bot instead of the regular message spammer.

This bot is not for malicious intent and is only for testing purposes. This can be bannable on discord if you use this for anything malicous. Use at your own risk.

This is best use for getting xp or leveling up in a server if they allow spamming. This can run in the background so feel free to do whatever you want on discord while it is running.

This is a discord nuker bot. It is very easy to use. I will provide a tutorial on how to get the channel id and authorization. Getting url and authorization requires web version of discord.

Requirments:
Download python at https://www.python.org/downloads/. Download latest version and install. Make sure to add python to path in the installation menu or it won't work. After you install python open up. You also need pip I think, correct me if I'm wrong in issues. If you do need it then go to https://www.geeksforgeeks.org/how-to-install-pip-on-windows/ and that will show you how.

Tutorial:

How to get channel-id?
First go onto the web version of discord and then go to the channel you would like to start spamming in. Look at the url of the channel, it should like like https://discord.com/channels/NUMBERS/channel-id.
You need to copy NUMBERS2, the very last numbers of the link. Paste those numbers in config on the url section. Where it says CHANNEL-ID-HERE, replace that text with the numbers.

How to get authorization?
First you need to go onto any channel you want. Open inspect element and go to network. After that send a random message in the server and you should see something called "messages" appear in the network. If you don't see it or the messages won't send to the channel just wait a second or retry to send the message until you see "messages". Click on messages and and you should be under headers. If you aren't then make sure you are. Scroll down until you see Request Headers and after you find that you will see something called authorization not to far under that. Copy the authorization message and replace it there it says "DISCORD AUTHORIZATION TOKEN HERE". Make sure to save the config and you should be done. double click on "run" and it should start. You need the python installed into path.

If you are confused on how to do it, join my discord and I'll tell you the steps on how to get everything running.


CHANGELOG
- Added auto refresh in case the bot freezes
- Massive speed increase. Many rate limits when fast so not the best. (Do not use for a long amount of time, discord only allows 10,000 requests every 10 minutes.
After using for at least 1 minute if you use the fast option then wait 10 minutes before using again.
If you do accidently reach the 10,000 requests limit you will be blocked from accessing discord for only 30m-1 hour)


discord-nuker/discordspam.py
import requests
import time
import logging
import os
import json
import datetime
from colorama import init, Fore

# Initialize colorama
init()

# Read settings from config.json file
with open("config.json") as f:
    SETTINGS = json.load(f)

# Set up logging
logging.basicConfig(filename='logs.log', level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')

discord_settings = SETTINGS.get("discord", {})
if discord_settings:
    message_list = discord_settings.get("message_list")
    authorization = discord_settings.get("authorization")
else:
    print(f"{Fore.RED}Error: 'discord' section missing from config file.{Fore.RESET}")
    logging.error("'discord' section missing from config file.")
    exit()

if not message_list or not authorization:
    print(f"{Fore.RED}Error: Discord 'message_list' and 'authorization' values must be specified in config file.{Fore.RESET}")
    logging.error("Discord 'message_list' and 'authorization' values must be specified in config file.")
    exit()

delay = SETTINGS.get("delay", 1)  # Default delay time if no delay specified in config
cooldown_time = SETTINGS.get("cooldown_time", 60)
message_count = 0  # Counter for number of messages sent
failed_urls = [] # List to store failed URLs
start_time = time.time() # Store start time for runtime display

try:
    while True:
        for message_obj in message_list:
            url = message_obj["url"]
            message = message_obj["message"]

            payload = {
                'content': message
            }

            r = requests.post(url, data=payload, headers={'authorization': authorization})
            if r.status_code == 200:
                success_message = f"{Fore.BLUE}Message '{message}' successfully sent to '{url}'{Fore.RESET}"
                print(success_message)
                logging.info(success_message)
            else:
                error_message = f"{Fore.RED}Error sending message '{message}'{Fore.RESET}"
                print(error_message)
                failed_urls.append(url)
                logging.error(f"{error_message} to '{url}'")

            message_count += 1
            if message_count % 10 == 0:
                # Clear console after every 10 messages
                os.system('cls' if os.name == 'nt' else 'clear')
                print(f"{Fore.GREEN}Running.{Fore.RESET}\n")

        if message_count % 50 == 0:
            print("Pausing for cooldown...")
            for i in range(cooldown_time, 0, -1):
                print(f"{Fore.YELLOW}\rResuming message sending in {i} seconds...", end="")
                time.sleep(1)
            print("\n") 

        time.sleep(delay)  # Rate limit delay

except Exception as e:
    error_message = f"{Fore.RED}Encountered an error: {e}{Fore.RESET}"
    print(error_message)
    logging.error(f"Encountered an error: {e}")
finally:
    if failed_urls:
        failed_message = f"{Fore.RED}The following URLs failed to send messages: {', '.join(failed_urls)}{Fore.RESET}"
        print(failed_message)
        logging.error(failed_message)

    runtime = time.time() - start_time
    print(f"Session runtime: {datetime.timedelta(seconds=runtime)}")

    print(f"{Fore.YELLOW}Discord message sender stopped.{Fore.RESET}")
    logging.info("Discord message sender stopped.")


discord-nuker/main.py
#Made by phantomonic#7836

import requests
import time
import logging
import os
import json
import datetime
import concurrent.futures
import asyncio
from colorama import init, Fore

# Initialize colorama
init()

# Read settings from config.json file
with open("config.json") as f:
    SETTINGS = json.load(f)

# Set up logging
logging.basicConfig(filename='logs.log', level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')

discord_settings = SETTINGS.get("discord", {})
if discord_settings:
    message_list = discord_settings.get("message_list")
    authorization = discord_settings.get("authorization")
else:
    print(f"{Fore.RED}Error: 'discord' section missing from config file.{Fore.RESET}")
    logging.error("'discord' section missing from config file.")
    exit()

if not message_list or not authorization:
    print(f"{Fore.RED}Error: Discord 'message_list' and 'authorization' values must be specified in config file.{Fore.RESET}")
    logging.error("Discord 'message_list' and 'authorization' values must be specified in config file.")
    exit()

delay = SETTINGS.get("delay", 1)  # Default delay time if no delay specified in config
cooldown_time = SETTINGS.get("cooldown_time", 60)
message_limit = SETTINGS.get("message_limit", 50)
send_count = SETTINGS.get("send_count", 1)
message_count = 0  # Counter for number of messages sent
failed_urls = [] # List to store failed URLs
start_time = time.time() # Store start time for runtime display
concurrency = SETTINGS.get("concurrency", 10)  # Number of concurrent requests


def send_message(url, message, authorization):
    payload = {
        'content': message
    }
    response = requests.post(url, data=payload, headers={'authorization': authorization})
    if response.status_code == 200:
        success_message = f"{Fore.BLUE}Message '{message}' successfully sent to '{url}'{Fore.RESET}"
        print(success_message)
        logging.info(success_message)
    else:
        error_message = f"{Fore.RED}Error sending message '{message}'{Fore.RESET}"
        print(error_message)
        failed_urls.append(url)
        logging.error(error_message)


async def dispatch_messages(message_list):
    with concurrent.futures.ThreadPoolExecutor(max_workers=concurrency) as executor:
        loop = asyncio.get_running_loop()
        futures = []
        for message_obj in message_list:
            for _ in range(send_count):
                url = message_obj["url"]
                message = message_obj["message"]
                futures.append(loop.run_in_executor(
                    executor,
                    send_message,
                    url,
                    message,
                    authorization
                ))
        await asyncio.gather(*futures)


async def main():
    global message_count
    while True:
        await dispatch_messages(message_list)
        message_count += len(message_list) * send_count
        if message_count % message_limit == 0:
            print(f"Pausing for {cooldown_time} seconds...")
            for i in range(cooldown_time, 0, -1):
                print(f"{Fore.YELLOW}\rResuming message sending in {i} seconds...", end="")
                await asyncio.sleep(1)
            print("\n")
        if message_count % (message_limit // 5) == 0:
            # Clear console after every 10% of message limit
            os.system('cls' if os.name == 'nt' else 'clear')
            print(f"{Fore.GREEN}Running.{Fore.RESET}\n")
        await asyncio.sleep(delay)  # Rate limit delay


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    try:
        loop.run_until_complete(main())
    except Exception as e:
        error_message = f"{Fore.RED}Encountered an error: {e}{Fore.RESET}"
        print(error_message)
        logging.error(f"Encountered an error: {e}")
    finally:
        if failed_urls:
            failed_message = f"{Fore.RED}{len(failed_urls)} messages failed to send.{Fore.RESET}"
            print(failed_message)
            logging.error(failed_message)

        runtime = time.time() - start_time
        print(f"Session runtime: {datetime.timedelta(seconds=runtime)}")

        print(f"{Fore.YELLOW}Nebula Nuker stopped.{Fore.RESET}")
        logging.info("Nebula Nuker stopped.")
